Makefile

```diff
@@ -1,10 +1,7 @@
-# get Makefile directory name: http://stackoverflow.com/a/5982798/376773
-THIS_MAKEFILE_PATH:=$(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
-THIS_DIR:=$(shell cd $(dir $(THIS_MAKEFILE_PATH));pwd)
+SINGLE_QUOTE := '
+# We need to escape single quote as '\''
+THIS_DIR := '$(subst $(SINGLE_QUOTE),$(SINGLE_QUOTE)\$(SINGLE_QUOTE)$(SINGLE_QUOTE),$(CURDIR))'
```

Comment:
True, but those Makefiles are independent of this one. So running `make test` directly from the directory with the tests obviously works as it should. As for `make test` from the root - that actually calls `make -C /path/to/test` - so, again, only the local Makefile will be used.
Just to confirm, I've tried both `make test` from the root directory to run all tests and `make test` in a subdirectory where some tests lie - both cases are working. But thanks for reminding me of this use case - it's good to double check stuff like that :)